Brokerages/DefaultOrderBook.cs

```diff
@@ -57,10 +65,10 @@ public class OrderBook
         public decimal BestAskSize { get; private set; }
```

Comment:
Are there any concerns regarding read consistency of the best bid/ask properties? For example, it is currently possible for one thread to be reading the bid/ask while another thread is updating them, leading to a potentially inconsistent read of the data (for example, an old bid and a new ask).

Reply:
I've been investigating this problem and found that we might want to use standard practice to avoid race condition and use (`lock(_locker)`) within get/set methods. I think it's good enough as our Properties are of value type. 
```
private int _myField;
private object _locker= new object();

public int MyProperty
{
    get
    {
        lock (_locker)
        {
            return _myField;
        }
    }
    set
    {
        lock (_locker)
        {
            _myField = value;
        }
    }
}
```

Any thoughts?

Reply:
I agree that read access to these properties should be synchronized, maybe it would be better to have only public getters with `lock` and the update methods can just update the backing fields (also with `lock`):

``` C#
private decimal _bestBidPrice;

public decimal BestBidPrice 
{ 
    get 
    {
        lock (_locker)
        {
            return _bestBidPrice;
        }
    }
}
```